### DISCUSSION TITLE
Message behaviour changed from 2.3.0 to 3.0.0

### DISCUSSION THREAD

**Discussion by azido** [nhielost/obs-midi-mg#90] [General]
Reproduceable: yes

What did i do:
Created a binding that changes the volume of my microphone in OBS via Midi.

Steps taken:
1. Added a new binding, named it Mic Volume OBS
2. Selected the device by using edit devices
3. Created a message with the following data:
- Type=Control Change
- Channel=1
- Control #=25
- Value MIDI 0-127 
4. Created an action with the following data:
- Category Audio Sources
- Action Change Source Volume
- Audio Source 1 Microfone
- Volume (db) Midi 0-127

When i did this on 2.3.0. all was working as expected. cc value 0 corresponded to the lowest slider position (-inf dB), 127 to the highest (0dB), all between moved the slider accordingly.
When i do this on 3.0.0, i get the following effect: cc value 0-3 do not change anything in OBSs slider, 4 "jumps" to -95 dB, 
![lowmidi](https://github.com/nhielost/obs-midi-mg/assets/151363/2bfc95a6-109c-4ac3-bec7-b0001f0a6822)
cc value 100 is at +0,2 db while the visual slider is on the top end
![highmidi](https://github.com/nhielost/obs-midi-mg/assets/151363/a4b25266-a395-4ce7-8d7e-feeeb845fab9)
and every cc value above 100 goes even higher until it reaches +27 dB ! without any visible slider effect in OBS
![maxmidi](https://github.com/nhielost/obs-midi-mg/assets/151363/e90729ed-afa9-40f3-b87c-ae525871b3b6)
(I guess this is the maximum value one can set for volume)

While i welcome the behaviour of going above 0 dB via Midi now (giving a lot more headroom to my button) the visual presentation of that slider does no longer reflect my knobtwists accordingly. If i want the old behaviour back, i have to change the Volume dB settings to custom and use - 99 / +0.

Is this intended?

**Comment by nhielost** [comment]
In terms of the decibel levels, the behavior you are describing is intended. However, there is nothing that I can do to increase the sliders visually past 0.0 dB in OBS itself, so this is the only way to accomplish higher volumes. 

If you would like, I can reintroduce the volume percentages from the older versions as a secondary option to the decibels. This would achieve the 0-100% slider visual you desire without the custom option, but this would be a tradeoff.  In percentage mode, it shows on the slider correctly, but it is impossible to go over 0.0 dB. In dB mode, volumes can go over 0.0 dB, but it does not show visually in OBS. 

Also, thanks for showing the 0.2 dB issue as I did not encounter it in testing.

**Comment by nhielost** [comment]
This discussion is being closed due to the implementation of the issue in question in version 3.0.0.
